### PR TITLE
Fix DataloaderHparam docs

### LINKED ISSUE #509: Incorrect setting of persistent_workers hparam
DataloaderHparams:
`persistent_workers: bool = hp.optional("Whether to shutdown workers after the dataset has been consumed once",                                              default=True)`

This makes it sound like the default option which is True shuts down the workers after the dataset has been consumed once. But when calling torch Dataloader, the default is False and this option keeps the workers alive.

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
Most importantly, clarified the meaning of `persistent_workers` in the dataloader hparams.

This PR has some unrelated style fixes caught by `make style` since `make lint` doesn't seem to be picking up on everything.

Closes #509

### CODE REVIEW

**Review by hanlint — APPROVED** [review]

File: composer/datasets/dataloader.py [hunk 1]
```diff
@@ -94,22 +94,42 @@ class DataloaderHparams(hp.Hparams):
...
 
     Parameters:
-        num_workers (int): Number of CPU workers to use per device to fetch data.
-        prefetch_factor (int): Number of samples loaded in advance by each worker.
-            2 means there will be a total of 2 * num_workers samples prefetched across all workers.
-        persistent_workers (bool): Whether or not to shutdown workers after the dataset has been consumed once.
-        pin_memory (bool): Whether or not to copy Tensors into CUDA pinned memory before returning them.
-        timeout (float): Timeout, in seconds, for collecting a batch from workers. Set to 0 for no timeout.
+        num_workers (int, optional): Number of CPU workers to use per device to fetch data.
+            Set to ``0`` to use the main training thread will be used for dataloading.
```
Comment (jfrankle) [inline composer/datasets/dataloader.py:98]: Grammatical mistake.

File: composer/datasets/dataloader.py [hunk 2]
```diff
@@ -94,22 +94,42 @@ class DataloaderHparams(hp.Hparams):
...
+            While zero workers can be useful for debugging, it should not be used for performance reasons.
+            (default: ``8``)
+        prefetch_factor (int, optional): Number of samples loaded in advance by each worker.
```
Comment (jfrankle) [inline composer/datasets/dataloader.py:101]: Is this samples or batches? I seem to recall it being batches in my past experience with PyTorch, but worth verifying.

File: composer/datasets/dataloader.py [hunk 3]
```diff
@@ -94,22 +94,42 @@ class DataloaderHparams(hp.Hparams):
...
+        timeout (float): Timeout, in seconds, for collecting a batch from workers. Set to ``0`` for no timeout.
+            (default: ``0``)
     """
 
-    num_workers: int = hp.optional("Number of CPU workers to use per device to fetch data.", default=8)
-    prefetch_factor: int = hp.optional("Number of samples loaded in advance by each worker", default=2)
-    persistent_workers: bool = hp.optional("Whether to shutdown workers after the dataset has been consumed once",
+    num_workers: int = hp.optional(textwrap.dedent("""\
+        Number of CPU workers to use per device to fetch data.
+        Set to ``0`` to use the main training thread will be used for dataloading.
```
Comment (jfrankle) [inline composer/datasets/dataloader.py:115]: Same grammatical error

**Review by jfrankle — CHANGES_REQUESTED** [review]
One question and one grammatical error.

File: composer/datasets/dataloader.py (hunk 2 above)
Comment (ravi-mosaicml) [reply composer/datasets/dataloader.py:101]: The docs here say samples: https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader

**Comment by ravi-mosaicml** [comment]
Red button merging as all jenkins failures were due to Jenkins storage issues, changes have been addressed, and all tests passed before.

**Pull request merged**